runtime/common/src/migrations.rs

```diff
@@ -0,0 +1,138 @@
+use frame_support::{
+    pallet_prelude::Weight,
+    traits::{Currency, OnRuntimeUpgrade, ReservableCurrency},
+};
+use frame_system::AccountInfo;
+use gear_common::{storage::LinkedNode, GasPrice, GasProvider, GasTree, Origin};
+use gear_core::ids::ProgramId;
+use sp_runtime::traits::{Get, UniqueSaturatedInto, Zero};
+#[cfg(feature = "try-runtime")]
+use sp_std::vec::Vec;
+
+type AccountIdOf<T> = <T as frame_system::Config>::AccountId;
+type Balances<T> = pallet_balances::Pallet<T>;
+type GearGas<T> = pallet_gear_gas::Pallet<T>;
+type GearBank<T> = pallet_gear_bank::Pallet<T>;
+type GasHandlerOf<T> = <GearGas<T> as GasProvider>::GasTree;
+type GasNodesOf<T> = pallet_gear_gas::GasNodes<T>;
+type AccountsOf<T> = frame_system::Account<T>;
+type CurrencyOf<T> = <T as pallet_gear_bank::Config>::Currency;
+type BalanceOf<T> = <CurrencyOf<T> as Currency<AccountIdOf<T>>>::Balance;
+type DispatchesOf<T> = pallet_gear_messenger::Dispatches<T>;
+type MailboxOf<T> = pallet_gear_messenger::Mailbox<T>;
+type WaitlistOf<T> = pallet_gear_messenger::Waitlist<T>;
+type DispatchStashOf<T> = pallet_gear_messenger::DispatchStash<T>;
+
+pub struct MigrateToGearBank<T, P>(sp_std::marker::PhantomData<(T, P)>)
+where
+    T: frame_system::Config<AccountData = pallet_balances::AccountData<BalanceOf<T>>>
+        + pallet_balances::Config<Balance = BalanceOf<T>>
+        + pallet_gear_gas::Config
+        + pallet_gear_bank::Config
+        + pallet_gear_messenger::Config,
+    P: GasPrice<Balance = BalanceOf<T>>,
+    AccountIdOf<T>: Origin;
+
+impl<T, P> OnRuntimeUpgrade for MigrateToGearBank<T, P>
+where
+    T: frame_system::Config<AccountData = pallet_balances::AccountData<BalanceOf<T>>>
+        + pallet_balances::Config<Balance = BalanceOf<T>>
+        + pallet_gear_gas::Config
+        + pallet_gear_bank::Config
+        + pallet_gear_messenger::Config,
+    P: GasPrice<Balance = BalanceOf<T>>,
+    AccountIdOf<T>: Origin,
+{
+    #[cfg(feature = "try-runtime")]
+    fn pre_upgrade() -> Result<Vec<u8>, &'static str> {
+        Ok(Default::default())
+    }
+
+    fn on_runtime_upgrade() -> Weight {
+        let version = T::Version::get().spec_version;
+
+        log::info!("🚚 Running migration to gear-bank with current spec version {version:?}");
+
+        // TODO: consider version here.
```

Comment:
when this TODO should be done?

Reply:
Once we decide when will we perform runtime upgrade containing this migrations